sycl/source/exception.cpp

```diff
@@ -103,7 +103,19 @@ __SYCL_EXPORT const char *stringifyErrorCode(int32_t error) {
     _UR_ERRC(UR_RESULT_ERROR_INVALID_DEVICE_PARTITION_COUNT)
     _UR_ERRC(UR_RESULT_ERROR_INVALID_WORK_ITEM_SIZE)
     _UR_ERRC(UR_RESULT_ERROR_INVALID_WORK_DIMENSION)
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#else
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
     _UR_ERRC(UR_RESULT_ERROR_INVALID_KERNEL_ARGS)
+#ifdef _MSC_VER
+#pragma warning(pop)
+#else
+#pragma GCC diagnostic pop
+#endif
```

Comment:
Instead of adding these pragma's to ignore the deprecation warnings, lets remove the use of the deprecated error code.